pkg/controller/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -443,60 +416,20 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileCredentialsRequest) usePassthroughCreds(cr *minterv1.CredentialsRequest, cloudCredSecret *corev1.Secret, logger log.FieldLogger) (*minterv1.CredentialsRequest, error) {
-	secretAlreadyExists := true
-	existingSecret := &corev1.Secret{}
-	err := r.Get(context.Background(), types.NamespacedName{Name: cr.Spec.SecretRef.Name, Namespace: cr.Spec.SecretRef.Namespace}, existingSecret)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			logger.Debug("existing secret not found. will create one.")
-			secretAlreadyExists = false
-		} else {
-			logger.WithError(err).Error("error checking if the secret satisfying the credentialsrequest already exists")
-			return nil, err
-		}
-	}
-
-	if secretAlreadyExists {
-		logger.Debug("Update if necessary")
-
-		if updateExistingSecret(existingSecret, cloudCredSecret) {
-			logger.Debugf("updating existing secret")
+func (r *ReconcileCredentialsRequest) updateActuatorConditions(cr *minterv1.CredentialsRequest, reason minterv1.CredentialsRequestConditionType, conditionError error) {
 
-			err = r.Update(context.Background(), existingSecret)
-			if err != nil {
-				logger.WithError(err).Error("error updating existing secret")
-				return nil, err
-			}
-		}
-	} else {
-		logger.Debug("Creating a secret")
-
-		newSecret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      cr.Spec.SecretRef.Name,
-				Namespace: cr.Spec.SecretRef.Namespace,
-				Annotations: map[string]string{
-					minterv1.AnnotationCredentialsRequest: fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
-				},
-			},
-			Data: map[string][]byte{
-				secretannotator.AwsAccessKeyName:       cloudCredSecret.Data[secretannotator.AwsAccessKeyName],
-				secretannotator.AwsSecretAccessKeyName: cloudCredSecret.Data[secretannotator.AwsSecretAccessKeyName],
-			},
-		}
+	// first unset any previous conditions
+	setInsufficientCredsCondition(cr, false)
+	setFailedToProvisionCredentialsRequest(cr, false, nil)
```

Comment:
Nervous about this section, if we were to get more fine grained with credentials conditions we might have some trouble maintaining this. Probably ok for now though.

Reply:
this was a weird one. the situation i had in mind when i structured it this way was as follows.

1) attempt to actuator.sync() fails b/c Reason/ConditionA
2) attempt to actuator.sync() fails b/c Reason/ConditionB
3) attempt to actuatior.sync() works

At the end of 3, which conditions should be set/cleared?

The approach I took was compiling a list of the conditions that actuator.sync() can return, and before setting any condition related to an error from the actuator.sync() call, just clear all actuator.sync()-possible conditions then set the one condition that is actually in the error we're presently handling.

So we're basically saying that any conditions that can be set as a result of actuator.sync() are exclusively setable only as a result of an actuator.sync() call (ie. credentials_request controller shouldn't set/unset these unless it's due to the results of an actuator.sync() call).